src/averageAges.js

```diff
@@ -55,9 +66,32 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const findAllWomenWithSon = (person) => {
+    return person.sex === 'm'
+    && people.some((mom) => mom.name === person.mother);
+  };
+
+  const findAllWomenWithChild = (person) =>
+    people.some((mom) => mom.name === person.mother);
+
+  const children = people.filter(
+    onlyWithSon ? findAllWomenWithSon : findAllWomenWithChild
+  );
+
+  const yearsOld = children.map(item => {
+    const mother = people.find(mom => item.mother === mom.name);
+
+    return item.born - mother.born;
+  });
+
+  return yearsOld.reduce((acc, el) => acc + el) / yearsOld.length;
 }
 
+const calculateAverageAge = (arr) => {
```

Comment:
Уникай в назвах змінних структур данних
```suggestion
const calculateAverageAge = (ages) => {
```